static/app/views/organizationGroupDetails/groupSimilarIssues/similarStackTrace/index.tsx

```diff
@@ -28,92 +30,31 @@ type Props = RouteComponentProps<RouteParams, {}> & {
   project: Project;
 };
 
-type State = {
-  error: boolean;
-  filteredSimilarItems: SimilarItem[];
-  loading: boolean;
-  similarItems: SimilarItem[];
-  similarLinks: string | null;
-  v2: boolean;
-};
-
-class SimilarStackTrace extends Component<Props, State> {
-  state: State = {
-    similarItems: [],
-    filteredSimilarItems: [],
-    similarLinks: null,
-    loading: true,
-    error: false,
-    v2: false,
-  };
-
-  componentDidMount() {
-    this.fetchData();
-  }
-
-  componentWillReceiveProps(nextProps: Props) {
-    if (
-      nextProps.params.groupId !== this.props.params.groupId ||
-      nextProps.location.search !== this.props.location.search
-    ) {
-      this.fetchData();
-    }
-  }
-
-  componentWillUnmount() {
-    this.listener?.();
-  }
-
-  onGroupingChange = ({
-    mergedParent,
-    similarItems,
-    similarLinks,
-    filteredSimilarItems,
-    loading,
-    error,
-  }) => {
-    if (similarItems) {
-      this.setState({
-        similarItems,
-        similarLinks,
-        filteredSimilarItems,
-        loading: loading ?? false,
-        error: error ?? false,
-      });
-      return;
-    }
-
-    if (!mergedParent) {
-      return;
-    }
+function SimilarStackTrace({params, location, project}: Props) {
+  const {orgId, groupId} = params;
 
-    if (mergedParent !== this.props.params.groupId) {
-      const {params} = this.props;
-      // Merge success, since we can't specify target, we need to redirect to new parent
-      browserHistory.push(
-        `/organizations/${params.orgId}/issues/${mergedParent}/similar/`
-      );
+  const [isUsingSimilarityViewV2, setIsUsingSimilarityViewV2] = useState<boolean>(false);
+  const [similarItems, setSimilarItems] = useState<SimilarItem[]>([]);
+  const [filteredSimilarItems, setFilteredSimilarItems] = useState<SimilarItem[]>([]);
+  const [similarLinks, setSimilarLinks] = useState<string | null>(null);
+  const [isLoading, setLoading] = useState<boolean>(true);
+  const [hasError, setHasError] = useState<boolean>(false);
```

Comment:
Probably outside the scope, but you can usually replace something like this with a single "status" type which makes calls to `this.setState({loading: true, error: false});` less error prone (in case you forget to clear other parts) as you can just call it as `this.setState({status: "loading"|"error"})`

Reply:
good call. I think it's down to three states: `'error' | 'loading' | 'ready'` and some inline vars can go away.